app/Console/Commands/CrawlPets.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Console\Commands;
+
+use App\Http\Integrations\Connectors\CrawlerConnector;
+use App\Http\Integrations\Requests\GetPageRequest;
+use App\Jobs\AnalyzePetPage;
+use App\Models\PetShelter;
+use App\Services\GeminiService;
+use App\Services\PetPageAnalyzer;
+use App\Services\PetShelterService;
+use App\Utils\DomAttributeExtractor;
+use App\Utils\UrlFormatHelper;
+use App\Utils\UrlValidator;
+use Illuminate\Console\Command;
+use Illuminate\Support\Facades\Log;
+use Saloon\Exceptions\Request\FatalRequestException;
+use Saloon\Exceptions\Request\RequestException;
+use Symfony\Component\DomCrawler\Crawler;
+
+class CrawlPets extends Command
+{
+    protected $signature = "crawl:pets {url?} {--depth=2} {--additional-urls=}";
+    protected $description = "Crawl saved url of shelter sites and analyze pages with AI to extract pet info";
+    protected PetShelterService $petShelterService;
+    protected GeminiService $gemini;
+    protected CrawlerConnector $connector;
+    protected PetPageAnalyzer $petPageAnalyzer;
+
+    public function handle(
+        GeminiService $gemini,
+        CrawlerConnector $connector,
+        PetPageAnalyzer $petPageAnalyzer,
+    ): void {
+        $this->gemini = $gemini;
+        $this->connector = $connector;
+        $this->petPageAnalyzer = $petPageAnalyzer;
+
+        $petSheltersWithExistingUrl = PetShelter::getAllPetShelterUrls();
+
+        if ($argumentUrl = $this->argument("url")) {
+            $petShelterUrls = collect([$argumentUrl]);
+        } else {
+            $petShelterUrls = collect($petSheltersWithExistingUrl);
+        }
+
+        if ($additionalUrls = $this->option("additional-urls")) {
+            $additionalUrlsArray = collect(explode(",", $additionalUrls))
+                ->map(fn($u) => trim($u))
+                ->filter()
+                ->unique()
+                ->diff($petShelterUrls);
+
+            $petShelterUrls = $petShelterUrls->merge($additionalUrlsArray);
+        }
+
+        $maxDepth = (int)$this->option("depth");
+
+        if ($maxDepth < 1) {
+            $this->error("Invalid depth value. It must be a positive integer.");
+
+            return;
+        }
+
+        foreach ($petShelterUrls as $index => $urlString) {
+            $this->info("Processing shelter " . ($index + 1) . " of " . count($petShelterUrls) . ": $urlString");
```

Comment:
That should be moved into Laravel Job/Queue and runned in new one process.

Here, u have dispatch of the job -> https://github.com/blumilksoftware/toby/blob/e086d8657a82017c8ca68af7faed1a06a4d489d5/app/Actions/VacationRequest/CancelAction.php#L27

And here is the job class implementation -> https://github.com/blumilksoftware/toby/blob/main/app/Jobs/ClearVacationRequestDaysInGoogleCalendar.php